lib/core/src/Cardano/Pool/DB.hs

```diff
@@ -166,6 +187,64 @@ data DBLayer m = forall stm. (MonadFail stm, MonadIO stm) => DBLayer
         -- For a Sqlite DB, this would be "run a query inside a transaction".
     }
 
+-- | Represents an abstract notion of a certificate publication time.
+--
+-- Certificates published at later times take precedence over certificates
+-- published at earlier times.
+--
+type CertificatePublicationTime = (SlotId, SlotInternalIndex)
```

Comment:
Consider making this into a `newtype`.

Reply:
I've promoted this type synonym into a concrete `data` type. (See 01fc73d8193597645069f1d1a4a5d7c0e79eeae2.)

Reply:
@rvl wrote:
> Can you find a less wordy name for it?

I can't (currently) think of a shorter name for this.

For example, we could use `SlotIndex`, but IMO that doesn't make it clear enough that the index is _internal_ to a slot.

Do you have any ideas for a better name?

Reply:
I've removed the `SlotInternalIndex` type, inlining the definition into `CertificatePublicationTime`. (See c7e5da754889c5ba684fbd82fa3d899996ff0624.)